### DISCUSSION TITLE
Can't get Netlify auth to work

### DISCUSSION THREAD

**Discussion by jerefrer** [majesticooss/astros#23] [Q&A] [answered]
Hi there, and thanks for your great work here.

I was trying to put together all the different pieces to have NetlifyCMS and i18n working at the same time when I found that you already did a wonderful job with this repo.

However although it works well on localhost I just can't understand how to setup the login for the `/admin` section so that it works after deploying on Netlify. By the way, I simply used the one-click deploy button you provide.

If I do as you said in the `README` and use `git-gateway` as `backend.name` then I end up in front of a username/password form.

If I understand correctly I have then to enable Netlify Identity on my Netlify app config page, then add a user there.

So I did that, and I received an e-mail with a confirmation link pointing to:
`https://my.app/#recovery_token=QUOSw1qqv91F_GDTgUVLig`

Obviously that can't work, and indeed when clicking on it I just end up on the home page.

Then I understood that in the config file I should comment or remove this line:
```
disableIdentityWidgetInjection: true,
```
(which would be nice to be mentioned in the `README` as well)

Then the confirmation/recovery links start to work, but when I try to login I get:
```
Git Gateway Error: Please ask your site administrator to reissue the Git Gateway token.
```
And after generating the token on Netlify, I get:
```
API_ERROR: Git Gateway Error: [object Object]
```
So still no luck ...

Failing to make it work this way, I tried to leave `github` as `backend.name` and update `repo`, `branch` and `base_url` to fit my project.

But then when I click "Log in with Github", I'm presented with this:
![image](https://github.com/zankhq/astros/assets/140954/d70a04cf-3cbb-4faf-b368-be73d0b06e25)

What I can't quite understand is that when I leave your original settings, then it seems to work since when I click "Log in with Github", I see this:
![image](https://github.com/zankhq/astros/assets/140954/16523bac-cf27-4e89-b38f-9edb3074d4ee)

I would try and click Authorize to see what happens after that but I obviously don't want to share access to all my repos.

I really don't know if that's me who did something wrong or you maybe missed some part of the Netlify-specific configuration.

Could you explain me in more details how to proceed ?
And maybe update the docs if indeed some steps are missing.

I had already tried to setup NetlifyCMS with a Hugo project and in the end I gave up and went with TinaCMS instead just because I couldn't figure out how to setup the admin login. I would be tremendously grateful if you could make all that clear to me ... :)

Thank you so much !

**Comment by zanhk** [comment, marked as answer]
Hi @jerefrer thanks for trying out astros.

I tried creating a brach feat/netlify and managed to make it work correctly with netlify, in the repo I also tried a new package so you should just take it as a reference and not clone from it. the backend configuration is that:


```
backend: {
					name: "git-gateway",
					repo: "zankhq/astros",
					branch: "feat/netlify",
					base_url: "https://dulcet-pothos-78fa4f.netlify.app",
},
```


change it accordingly to your infos (make sure to remove` auth_endpoint: "/api/auth"` as well (also make sure you are using the proper branch name master vs main).

then as you said I removed `disableIdentityWidgetInjection: true,`

In netlify then I did:

1. enabled netlify identity from https://app.netlify.com/sites/[xxx]/integrations/identity
![image](https://github.com/zankhq/astros/assets/17980844/27a73f4e-b9b2-4896-8b4a-e8bfb8857a70)
2. Invited myself 
![image](https://github.com/zankhq/astros/assets/17980844/e1e50945-b6e7-45b2-8dd3-b400123cc784)
3. Enabled git gateway from https://app.netlify.com/sites/[xxx]/configuration/identity#registration
![image](https://github.com/zankhq/astros/assets/17980844/323a69f4-9ac7-4eae-97e5-8cc0dab844bd)
4. Opened the email received for the invitation
5. Set the password
6. go mysite/admin
7. edit content

**Comment by jerefrer** [comment]
Alright ... Followed your steps and it worked !

I guess removing `auth_endpoint: "/api/auth"` did the trick.

Thank you so much for replying so quickly !

I guess you have now an almost ready-made Wiki page for future Netlify users of Astros :)
